### PR TITLE
ADR and article on media types and versioning

### LINKED ISSUE #149: Remove all `application/json` entries from requestBody elements
Also add an ADR for the conclusion that the API should not allow for the use of this generic media type due to concerns about how to manage versioning and the like

### PR DESCRIPTION
**Pull request by catkins-miso**

### CODE REVIEW

File: docs/articles/media-types.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+> (emphasis added)
+
+As an OpenAPI specification, TROLIE uses [JSON
+Schema](https://swagger.io/docs/specification/data-models/keywords/) to describe
+the data formats of [media
+types](https://swagger.io/docs/specification/media-types/), so to define
+processing models for these media types, TROLIE only needs to describe
+the semantics of the transmitted JSON. For the most part, the [TROLIE OpenAPI
+Specification](../spec) does a decent job of explaining what each of the JSON
+elements mean, but that can be unfamiliar to navigate. This article aims to give
```
Comment (getorymckeag) [inline docs/articles/media-types.md]:
```suggestion
the semantics of the transmitted JSON. While the [TROLIE OpenAPI
Specification](../spec) explains what each of the JSON elements mean, 
it assumes that the reader understands how to use media types in an
API. This article aims to give
```

Just my reaction as I read this; I don't think it read quite how you intended it to read.

File: docs/articles/media-types.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+### Data Submittal (Write Requests)
+
+The following media types are used when writing to a TROLIE server. In other
+words, one of these media types are required in the `Content-Type` header for
+HTTP the `PATCH` and `PUT` requests specified by TROLIE.
+
+For example, here is how you could use `curl` to submit forecasted ratings
+assuming the file `forecast.json` was a valid document per the
+`application/vnd.trolie.rating-forecast-proposal.v1+json` schema. TODO: links
```
Comment (getorymckeag) [inline docs/articles/media-types.md]: Clean out TODO:

**Review by getorymckeag — COMMENTED** [review]
This is great- after I did what I could best describe as a proof-read, I sat back and tried to think of broader context things you may have missed.  The only thing I could think of is a demonstration of the benefits of media types as an API versioning technique.  Specifically, illustrating that the API can change, and how that might get leveraged by clients while not affecting new features.  

Or maybe we can demonstrate this as well with the slim media types- the idea that we can easily support multiple formats to represent the same thing logically without breaking existing clients.  

I leave it up to you if you want to take that further- I will happily approve it in its current state as an improvement if you want to just do the proof-reading bits.

File: docs/articles/media-types.md (hunk 1 above)
Comment (catkins-miso) [reply docs/articles/media-types.md]: I believe I addressed this feedback https://github.com/trolie/spec/pull/161/files#diff-e217f8b4ad8a310b77585ef760bf29ae57912123ae78b8e802547d0b3a598540R35

**Review by getorymckeag — APPROVED** [review]

**Pull request merged**